src/node/pool.rs

```diff
@@ -0,0 +1,62 @@
+use crate::node::impersonate::ImpersonationManager;
+use std::sync::{Arc, RwLock};
+use zksync_types::l2::L2Tx;
+
+#[derive(Clone)]
+pub struct TxPool {
```

Comment:
Nice, this will enable us to add [`--order`](https://github.com/foundry-rs/foundry/blob/25cc1ac68b5f6977f23d713c01ec455ad7f03d21/crates/anvil/src/cmd.rs#L115)